R/aes.r

```diff
@@ -143,13 +143,37 @@ print.uneval <- function(x, ...) {
   new_aes(NextMethod())
 }
 
-# Rename American or old-style aesthetics name
-rename_aes <- function(x) {
-  # Convert prefixes to full names
-  full <- match(names(x), ggplot_global$all_aesthetics)
-  names(x)[!is.na(full)] <- ggplot_global$all_aesthetics[full[!is.na(full)]]
+#' Standardise aesthetic names
+#'
+#' This function standardises aesthetic names by converting `color` to `colour`
+#' (also in substrings, e.g. `point_color` to `point_colour`) and translating old style
+#' R names to ggplot names (eg. `pch` to `shape`, `cex` to `size`).
+#' @param x Character vector of aesthetics names, such as `c("colour", "size", "shape")`.
+#' @return Character vector of standardised names.
+#' @keywords internal
+#' @export
+standardise_aes_names <- function(x) {
+  # convert US to UK spelling of colour
+  x <- sub("color", "colour", x, fixed = TRUE)
 
-  plyr::rename(x, ggplot_global$base_to_ggplot, warn_missing = FALSE)
+  # convert old-style aesthetics names to ggplot version
+  plyr::revalue(x, ggplot_global$base_to_ggplot, warn_missing = FALSE)
+}
+
+# x is a list of aesthetic mappings, as generated by aes()
+rename_aes <- function(x) {
```

Comment:
Where does this get called from?

Reply:
The main user is the `aes()` function, but it's actually called from many places:
https://github.com/tidyverse/ggplot2/blob/17b45f906c3a55d187915f4f3010836a804051ae/R/aes.r#L77-L83
https://github.com/tidyverse/ggplot2/blob/528b0ce369df71dec2f7126e6a40640f025c390f/R/geom-defaults.r#L17
https://github.com/tidyverse/ggplot2/blob/429991746ef69d7d66b1b33bb4c24e89b521907f/R/labels.r#L57
https://github.com/tidyverse/ggplot2/blob/c1908f192db6f5b5937f38ae1f2fc0410e4be7d5/R/quick-plot.r#L86
https://github.com/tidyverse/ggplot2/blob/3129bf861dfbc4eeba6121eaef02dceabdb6dbda/R/guides-.r#L63
https://github.com/tidyverse/ggplot2/blob/17b45f906c3a55d187915f4f3010836a804051ae/R/layer.r#L99-L100
https://github.com/tidyverse/ggplot2/blob/17b45f906c3a55d187915f4f3010836a804051ae/R/guide-legend.r#L183